cpp/ql/src/semmle/code/cpp/models/implementations/Memcpy.qll

```diff
@@ -52,4 +53,20 @@ class MemcpyFunction extends ArrayFunction, DataFlowFunction, TaintFunction {
     ) and
     countParam = 2
   }
+  
+  override predicate neverWritesMemory() {
```

Comment:
`neverReadsMemory` and `neverWritesMemory` are a little confusing, since they don't actually seem to mean "never". How about something like `hasOnlySpecific[Read|Write]SideEffects`?